app/controllers/assignment_swap_controller.rb

```diff
@@ -0,0 +1,34 @@
+class AssignmentSwapController < SecureController
+
+  skip_before_action :authenticate_user!
+
+  def index
+    @weeks = current_office.weeks.upcoming 
+  end
+
+  def create
+    #no assignments were selected
+    if !(params.has_key?("assignment"))
+      redirect_to office_assignment_swap_index_path
+      return
+    end
+
+    # only 1 or more than 2 assignments were selected
+    if params[:assignments][:assignment_id].count != 2
+      redirect_to office_assignment_swap_index_path
+      return
+    end
+
+    ass1 = Assignment.find(params[:assignments][:assignment_id].first)
+    ass2 = Assignment.find(params[:assignments][:assignment_id].second)
+
+    employee1 = ass1.employee
+    employee2 = ass2.employee
+
+    ass1.update(employee: employee2)
+    ass2.update(employee: employee1)
```

Comment:
Whenever you're updating more than one database record each change depends on other changes being successful, you should use a transaction. ActiveRecord provides a fairly simple way of doing transactions:

```
Assignment.transaction do
  assignment1.update!(...)
  assignment2.update!(...)
end
```

Notice the `!`s. We need to raise errors if we want transactions to roll back. We could handle the exceptions (with `rescue`) but I don't foresee the exceptions happening under normal conditions so I would just let them go.